### ISSUE TITLE
 HighLevelProducer default key/message serialization

### ISSUE THREAD

**Issue by norla** [Blizzard/node-rdkafka#620]
**Environment Information**
 - node-rdkafka version [2.6.1]:

**Description**
HighLevelProducer default key/message serialization seems broken.

If you don't set message/key serializers nothing is sent and no errors are emitted.
I think it is because the default serializers are not fed through the "createSerializer" function which alters the behaviour of the "apply" function (it originally takes two arguments).

https://github.com/Blizzard/node-rdkafka/blob/50f99ab46c4a475678b94af07c2ea02e1ba353ff/lib/producer/high-level-producer.js#L259

⬆️ This returns "undefined" with javascript's built-in "apply" function.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/apply

**How to reproduce**

```
const Kafka = require("node-rdkafka");
const producer = new Kafka.HighLevelProducer({
  "metadata.broker.list": "localhost:9200"
});
producer.connect();
producer.on("ready", () => {
  producer.produce("some-topic", null, Buffer.from("Hello"), "some-key", Date.now(), (err, res) => {
    console.log("Message sent", err, res);
  });
});
```

**Comment by JaapRood** [comment]
There's 2 PRs, either of them should fix this #618, #592.

**Comment by grgrssll-dev** [comment]
in the mean time you can do this and it will work (as I documented in #600 )

```
producer.setValueSerializer(v => v);
producer.setKeySerializer(v => v);
```
